iio-widgets/src/datastrategy/cmdqchannelattrdatastrategy.cpp

```diff
@@ -97,9 +99,12 @@ void CmdQChannelAttrDataStrategy::attributeReadFinished(Command *cmd)
 		return;
 	}
 
-	m_dataRead = QString(tcmd->getResult());
+	QString newData = QString(tcmd->getResult());
 	if(!m_recipe.constDataOptions.isEmpty()) {
+		QString oldData = m_dataRead;
 		m_optionalDataRead = m_recipe.constDataOptions;
+		m_dataRead = newData;
+		Q_EMIT emitStatus(QDateTime::currentDateTime(), oldData, m_dataRead, tcmd->getReturnCode(), true);
 		Q_EMIT sendData(m_dataRead, m_optionalDataRead);
 	} else if(!m_recipe.iioDataOptions.isEmpty()) {
 		Command *readOptionalCommand = new IioChannelAttributeRead(
```

Comment:
On this branch `m_dataRead` is never updated and the `sendData` signal from `CmdQChannelAttrDataStrategy::optionalAttrReadFinished` emits inconsistent data.